### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by nim442**
Merging to master

### CODE REVIEW

File: UITest/UITestSample.cs [hunk 1]
```diff
@@ -43,67 +45,82 @@ public void BeforeEachTest()
...
         public void SelectPreferences()
         {
-            app.WaitForElement(c => c.Marked("button1"));
+            app.WaitForElement(c => c.Marked("loginButton"));
             app.Invoke("StartActivityOne");
             app.Tap(c => c.Marked("radioButtonNonSmoker"));
             app.Tap(c => c.Marked("radioButtonNoLuggage"));
             app.Tap(c => c.Marked("radioButtonNoHandicap"));
             app.Tap(c => c.Marked("radioButtonNoPet"));
             app.Tap(c => c.Marked("text1"));
```
Comment (AlineTheAlien) [inline UITest/UITestSample.cs]: We should change this id to make sure it's as specific as possible, and so we don't accidentally have more than 1 element with the same id

File: mRides-app/LeaveReviewFragment.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        public override void OnAttach(Activity activity)
+        {
+            base.OnAttach(activity);
+        }
+
+        // Load a new activity and transfer the data to the new one
+        void SendReviewButtonClicked(object sender, EventArgs e)
+        {            
+            //rideID : 2
```
Comment (AlineTheAlien) [inline mRides-app/LeaveReviewFragment.cs]: We should remove unnecessary comments before we forget

File: mRides-app/PreferencesActivity.cs [hunk 3]
```diff
@@ -98,30 +83,66 @@ protected override void OnCreate(Bundle savedInstanceState)
...
-            this.Continue();
-        }
+            // If the current user is null and the previous activity is the main, 
+            // create a new user using the preferences
+            string previousActivity = Intent.GetStringExtra(Constants.IntentExtraNames.PreviousActivity);
+            if (User.currentUser == null && previousActivity.Equals(Constants.ActivityNames.MainActivity)) {
+                long facebookID = Convert.ToInt64(Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookId));
+                string facebookFirstName = Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookFirstName);
+                string facebookLastName = Intent.GetStringExtra(Constants.IntentExtraNames.UserFacebookLastName);
+                User newUser = new User
```
Comment (AlineTheAlien) [inline mRides-app/PreferencesActivity.cs]: We can have a nondefault constructor of User that accepts all these fields, will make the code more compact and easier to read

File: UITest/UITestSample.cs (hunk 1 above)
Comment (nim442) [inline UITest/UITestSample.cs]: That would change our strings.xml also. I've made ad issue and assigned Hannah to it

File: mRides-app/Resources/layout/UserProfileFeedbackFragment.axml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:background="#FFFFFF"
+    android:minWidth="25px"
+    android:minHeight="25px"
+    android:id="@+id/linearLayout1">
```
Comment (AlineTheAlien) [inline mRides-app/Resources/layout/UserProfileFeedbackFragment.axml]: Once again, we should not have the default names for the element id's, they should be as specific as possible so we can never accidentally have duplicates

File: mRides-app/Resources/layout/UserProfileFragment.axml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content">
+    <LinearLayout
+        android:orientation="vertical"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:id="@+id/linearLayout2">
```
Comment (AlineTheAlien) [inline mRides-app/Resources/layout/UserProfileFragment.axml]: This id name should be changed too

File: mRides-app/Resources/layout/UserTypeFragment.axml [hunk 6]
```diff
@@ -2,40 +2,45 @@
...
+    android:background="#50000000">
     <LinearLayout
         android:orientation="horizontal"
-        android:layout_width="356.0dp"
+        android:layout_width="364.0dp"
         android:layout_height="61.0dp"
-        android:layout_margin="10dp">
+        android:layout_margin="10dp"
+        android:layout_marginRight="23.5dp"
+        android:id="@+id/linearLayout2">
```
Comment (AlineTheAlien) [inline mRides-app/Resources/layout/UserTypeFragment.axml]: Change id name

File: UITest/UITestSample.cs (hunk 1 above)
Comment (nim442) [inline UITest/UITestSample.cs]: Made Issue #33 to address this

File: mRides-app/UserProfileActivity.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            //fb.userIdForNow = 2;
+            //fb.username = "Nassim";
+
+            //FeedbackForNow fb2 = new FeedbackForNow();
+            //fb2.dateOfFeedback = DateTime.Now;
+            //fb2.rating = 5;
+            //fb2.review = "Cool";
+            //fb2.userIdForNow = 3;
+            //fb2.username = "Hannah";
+
```
Comment (AlineTheAlien) [inline mRides-app/UserProfileActivity.cs]: In the future, if commented code blocks aren't needed anymore, we should remove them

File: mRides-app/UserProfileFeedbackAdapter.cs [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        public UserProfileFeedbackAdapter(Activity activity)
+        {
+            context = activity;
+            FillFeedbacks();
+        }
+
+        void FillFeedbacks()
+        {
+
+        }
```
Comment (AlineTheAlien) [inline mRides-app/UserProfileFeedbackAdapter.cs]: empty unused method

File: mRides-app/PreferencesActivity.cs (hunk 3 above)
Comment (limphilip) [inline mRides-app/PreferencesActivity.cs]: That's a good idea, I will create an issue for that.

**Pull request merged**